board/app/controllers/thread_controller.php

```diff
@@ -1,107 +1,145 @@
 <?php
-class ThreadController extends AppController {
 
-    const MAX_THREAD_PER_PAGE = 7;
-    const MAX_COMMENT_PER_PAGE = 5;
-  
-    /*
-    * Create new thread
-    * :: - STATIC FUNCTION, can be called from the class name
-    * -> - INSTANCE, can only be called from an instance of the class.
-    */
-    public function create()
+class ThreadController extends AppController 
+{
+
+const CREATE_THREAD = 'create';
+const CREATE_THREAD_END = 'create_end';
+
+/*
+* Create new thread
+* :: - STATIC FUNCTION, can be called from the class name
+* -> - INSTANCE, can only be called from an instance of the class.
+*
+* Everything inputted on the view will be gathered by this function
+*/
+
+   public function create()
     {
         $thread = new Thread();
         $comment = new Comment();
-        $current_page = Param::get('page_next', 'create');
-      
-        switch ($current_page) {
-            case 'create':
-            break;
-      
-        case 'create_end':
-            $thread->title = Param::get('title');
-            $comment->username = Param::get('username');
-            $comment->body = Param::get('body');
- 
-            try 
-            {
-                $thread->create($comment);
-            } catch (ValidationException $e) {
-                $current_page = 'create';
-            }
-              break;
+        $current_page = Param::get(PAGE_NEXT, self::CREATE_THREAD);
+                    
+        switch ($current_page) { 
+            case self::CREATE_THREAD:
+                break;
+
+            case self::CREATE_THREAD_END:
+                $thread->title = Param::get('title'); 
+                $thread->category = Param::get('category');
+                $thread->user_id = $_SESSION['user_id'];
+                $comment->body = Param::get('body');
+                $comment->user_id = $_SESSION['user_id'];
+                $comment->username = $_SESSION['username'];
+          
+                try {
+                    $thread->create($comment);
+                } catch (ValidationException $e) {
+                    $current_page = self::CREATE_THREAD;
+                }
+                break;
+                
             default:
                 throw new NotFoundException("{$current_page} is not found");
-            break;
-        }  
+                break;
+            }  
```

Comment:
Done po. Checked my other codes as well. Thank you :bow:
--> https://github.com/maragatchalian/FoodFacts/commit/875a5bd857b0b0260ae1bb35a74cf4f7eae18b05